src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs

```diff
@@ -496,11 +496,24 @@ public WorkspaceViewModel(WorkspaceModel model, DynamoViewModel dynamoViewModel)
             };
 
             // sync collections
-            foreach (NodeModel node in Model.Nodes) Model_NodeAdded(node);
-            foreach (NoteModel note in Model.Notes) Model_NoteAdded(note);
-            foreach (AnnotationModel annotation in Model.Annotations) Model_AnnotationAdded(annotation);
-            foreach (ConnectorModel connector in Model.Connectors) Connectors_ConnectorAdded(connector);
-            
+            using (Errors.DeferCollectionReset())
+            using (Nodes.DeferCollectionReset())
+            {
+                foreach (NodeModel node in Model.Nodes) Model_NodeAdded(node);
```

Comment:
sadly we cannot simply use AddRange or RemoveRange here...because the binding WPF controls enforce the expectation that CollectionChanged of type Add or Remove should only have 1 changed item (not an array)
So the next best thing is to use Reset

Reply:
removed since it does not help performance that much in this case 
The WPF control (that binds to these collections) is not created yet (at this stage)